facia-press/app/frontpress/FapiFrontPress.scala

```diff
@@ -119,6 +131,38 @@ trait FapiFrontPress extends QueryDefaults with Logging with ExecutionContexts {
         backfill.map(slimContent),
         treats.map(slimContent))
 
+  private def getCurated(collection: Collection): Response[List[PressedContent]] = {
+    // Map initial PressedContent to enhanced content which contains pre-fetched embed content.
+    val initialContent = collectionContentWithSnaps(collection, searchApiQuery, itemApiQuery)
+    initialContent.flatMap(content => Response.traverse(content.map(fetchEmbeds)))
+  }
+
+  private def fetchEmbeds(pressed: PressedContent): Response[PressedContent] = pressed match {
+    case content: CuratedContent if FaciaInlineEmbeds.isSwitchedOn => {
+        val currentEnriched = content.enriched.getOrElse(EnrichedContent.empty)
+
+        val newContent = for {
+          embedType <- content.properties.embedType if embedType == "json.html"
+          embedUri <- content.properties.embedUri
+        } yield {
+            val updatedContent = WS.url(embedUri).get().map { response =>
+              Json.parse(response.body).validate[EmbedJsonHtml] match {
+                case JsSuccess(embed, _) => {
+                  val newEnriched = currentEnriched.copy(embedHtml = Some(embed.html))
+                  content.copy(enriched = Some(newEnriched))
+                }
+                case _ => content
+              }
+            } recover {
+              case _ => content
+            }
+            Response.Async.Right(updatedContent)
+          }
+        newContent.getOrElse(Response.Right(content))
+      }
+    case _ => Response.Right(pressed)
```

Comment:
We should log it, yes. The fallback behaviour is today's behaviour, so when original `PressedContent` is used, client-side javascript will enhance the content card.